src/Fahrplanauskunft/Fahrplanauskunft/Objekte/Haltestelle.cs

```diff
@@ -47,6 +47,10 @@ public override bool Equals(object obj)
             {
                 throw new NullReferenceException();
             }
+            if(!(obj is Haltestelle))
+            {
+                return false;
+            }
             Haltestelle other = obj as Haltestelle;
             return this.Name == other.Name;
```

Comment:
Die Equal-Methode hält sich nicht an die [Richtlinie von Microsoft](https://msdn.microsoft.com/de-de/library/ms173147(v=vs.90).aspx)

Mein Vorschlag:  
````c
public override bool Equals(object obj)     
{  
	if (obj == null)  
	{  
		return false;  
	}  
	Haltestelle other = obj as Haltestelle;  
	if (other == null)  
	{  
		return false;  
	}  
	return this.Name == other.Name;  
}
````
Die müsste auch in den anderen Klassen geändert werden